musicbot/bot.py

```diff
@@ -5545,6 +5545,43 @@ async def cmd_config(
                 f"To save the change use `config save {opt.section} {opt.option}`",
                 delete_after=30,
             )
+        
+        # reset an option to default value as defined in ConfigDefaults
+        if option == "reset":
+            if not opt.editable:
+                raise exceptions.CommandError(
+                    f"Option `{opt}` is not editable. Cannot reset to default.",
+                    expire_in=30,
+                )
+
+            # Use the default value from the option object
+            default_value = opt.default
+
+            # Handle different types of default values
+            if isinstance(default_value, set):
+                # If the default is a blank set, represent it as an empty string
+                default_value = ",".join(default_value) if default_value else ""  # type: ignore
+            elif isinstance(default_value, pathlib.Path):
+                default_value = str(default_value)
+            elif not isinstance(default_value, str):
+                default_value = str(default_value)
+
+            # Prepare a user-friendly message for the reset operation
+            reset_value_display = default_value if default_value else "an empty set"
```

Comment:
Not a change request, just a note.   There is also some not-quite-finished code related to display values in the option registry.  Most of that was added for the permissions side of things though.   Might need a little careful tweaking, but could be useful here too.